table/column.go

```diff
@@ -192,7 +192,7 @@ func CastValue(ctx sessionctx.Context, val types.Datum, col *model.ColumnInfo) (
 			}
 			casted, err = handleWrongUtf8Value(ctx, col, &casted, str, i)
 			break
-		} else if width > 3 && utf8Charset && config.GetGlobalConfig().CheckMb4ValueInUtf8 {
+		} else if width > 3 && utf8Charset && config.GetGlobalConfig().CheckMb4ValueInUtf8 && !config.GetGlobalConfig().IgnoreColumnUTF8Charset {
```

Comment:
We are only dealing with the inconsistency between the charset of table and column, and the table is the old version.